istioctl/pkg/install/pre-check.go

```diff
@@ -171,21 +187,32 @@ func installPreCheck(istioNamespaceFlag string, restClientGetter resource.RESTCl
 		fmt.Fprintf(writer, "This Kubernetes cluster deployed without MutatingAdmissionWebhook support."+
 			"See https://istio.io/docs/setup/kubernetes/additional-setup/sidecar-injection/#automatic-sidecar-injection\n")
 	} else {
-		fmt.Fprintf(writer, "Istio Configured to use automatic sidecar injection for 'default' namespace."+
-			" To enable injection for other namespaces see https://istio.io/docs/setup/kubernetes/additional-setup/sidecar-injection/#deploying-an-app\n")
+		fmt.Fprintf(writer, "This Kubernetes cluster supports automatic sidecar injection."+
+			" To enable automatic sidecar injection see https://istio.io/docs/setup/kubernetes/additional-setup/sidecar-injection/#deploying-an-app\n")
 	}
 	fmt.Fprintf(writer, "\n")
 	fmt.Fprintf(writer, "-----------------------\n")
-	fmt.Fprintf(writer, "Install Pre-Check passed! The cluster is ready for Istio installation.\n")
+	if errs == nil {
+		fmt.Fprintf(writer, "Install Pre-Check passed! The cluster is ready for Istio installation.\n")
+	}
 	fmt.Fprintf(writer, "\n")
-	return nil
+	return errs
 
 }
 
-func checkKubernetesVersion(versionInfo *version.Info) bool {
-
-	v := strings.Join([]string{versionInfo.Major, versionInfo.Minor}, ".")
-	return parseVersion(minK8SVersion, 4) < parseVersion(v, 4)
+func checkKubernetesVersion(versionInfo *version.Info) (bool, error) {
```

Comment:
Consider splitting the code into two methods, one that extracts the version from versionInfo.GitVersion and one that does the comparison.

I say this because it has been my experience that future developers modify the version-extracting regexp and break things unless there are tests for version string parsing.

Reply:
done